### PR TITLE
[WIP] Puppeteer

### PR DESCRIPTION
**Pull request by jyaworski**
Per https://github.com/pmd/pmd and https://github.com/tushartushar/Puppeteer

### CODE REVIEW

**Comment by jyaworski** [comment]
I feel I'm abusing fixtures.yml....

Does anyone know how to just download and extract a file inside of travis?

@nibalizer @igalic

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#  Copyright 2015 Puppet Community
+#  Copyright 2016 VoxPupuli
```
Comment (daenney) [inline Gemfile]: Spaces! Vox Pupuli, or voxpupuli when we're talking username/namespace.

**Comment by daenney** [comment]
This PR needs at least some explanation of what you're trying to do :stuck_out_tongue:.

**Comment by jyaworski** [comment]
@daenney sorry, I should have specified. Puppeteer is sort of like puppet-lint and rubocop, in that it warns you about code smells. @igalic mentioned it, and I decided to look into it. This would be another linter that we would add to our matrix.

File: .fixtures.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+fixtures:
+  repositories:
+    pmd:
+      repo: 'git://github.com/pmd/pmd.git'
+      ref: 'pmd_releases/5.4.1'
+    puppeteer:
+      repo: 'git://github.com/tushartushar/Puppeteer'
```
Comment (igalic) [inline .fixtures.yml]: i'd prefer https:// here, because some poor folks might be sitting behind weird firefalls

Comment (jyaworski) [inline .fixtures.yml]: I'm still wondering if I should even keep this in fixtures.

**Comment by jyaworski** [comment]
I'm closing this until I think a lot more on it.

**Comment by igalic** [comment]
yah, thinking good.

**Pull request closed without merging**